### ISSUE TITLE
Compatibility Issues with urql@1.8.1 and wonka@3.2.2

### ISSUE THREAD

**Issue by papigers** [FormidableLabs/urql#505]
**Describe the bug**
I a relatively new project I'm working on I added `urql` and `wonka` to my project, in order to use urql with a custom exchange.
these were the versions:
```
    "urql": "^1.7.0",
    "wonka": "^3.2.2",
```
This week I've upgraded my dependencies, and for some reason, only urql was updated to 1.8.1 and wonka was not.
After that I've encountered issues with performing queries (tried both the useQuery hook and the component) since I'm developing on react-native, debugging was extremely difficult.
After much debugging, I've reached to the wonka code that is reached at the [this](https://github.com/FormidableLabs/urql/blob/master/src/hooks/useQuery.ts#L81) scan call (not sure exactly where), when I eventually gave up, since the code was minified at react-native-dev-tools and didn't make much sense to me.
Thankfully, I've read this line from the Releases page:

> If you're using wonka for an exchange with urql you may want to upgrade to wonka@^4.0.5 soon.

And indeed after upgrading, everything was working again.
But the same paragraphs claim the upgrade isn't necessary when in fact it is.

It is also important to note that if wonka is not used there isn't any problem - it's only a compatibility issue.

**Steps to reproduce**
I've made this [codesandbox ](https://codesandbox.io/s/xenodochial-sun-0q986)which fails to run (fails downloading `wonka`), but the code itself should not work with `wonka@^3.2.2` and `urql@^1.8.1`, upgrading to `wonka@^4.0.5` does the trick.

**Expected behavior**
Queries should be working

**Actual behavior**
Queries are not.

**Additional context**
Above.

**Comment by kitten** [comment]
So two things may have happened:

- either your package manager has ignored urql’s Version requirement for Wonka and is only using the old one (you can check that with npm/yarn list)
- or your custom exchange has a bug or edge case that has only surfaced due to mixed versions

However, I wouldn’t see this as saying “it’s unsafe to upgrade with these instructions”. The urql package has the new Wonka dependency and our change log hints at this so unfortunately this is all we can do to tell people how to upgrade 😅

**Comment by papigers** [comment]
- Possible but unfortunately, I've already upgraded wonka to v4 so the output of `npm list` won't do much help. I'm adding my `package-lock.json` diffs if you want to check what may have happened, but I'm fine with just closing.
- I already attempted removing the exchange from the client exchange list, I've also attempted removing the graph cache exchange, and using the default exchanges, since I've become so desperate understanding where the issue comes from. It's not about that - installing the wonka package itself is what caused the issue.

**package-lock.json diff:**
![image](https://user-images.githubusercontent.com/5209100/72264305-32aa5380-3623-11ea-8550-417f677ca250.png)
![image](https://user-images.githubusercontent.com/5209100/72264329-405fd900-3623-11ea-8fb0-d2131ff8ae62.png)
![image](https://user-images.githubusercontent.com/5209100/72264386-566d9980-3623-11ea-8942-3df4ae0dc768.png)
![image](https://user-images.githubusercontent.com/5209100/72264408-65544c00-3623-11ea-8e03-791cb89fda49.png)
